panel/tests/ui/test_custom.py

```diff
@@ -0,0 +1,357 @@
+      model.on("change:text", cb2)
+      el.append(h1, h2)
+    }
+    """
+
+@pytest.mark.parametrize('component', [JSUnwatch, AnyWidgetUnwatch])
+def test_unwatch(page, component):
+    example = component(text='Hello World!')
+
+    serve_component(page, example)
+
+    expect(page.locator('h1')).to_have_text('Hello World!')
+    expect(page.locator('h1')).to_have_text('Hello World!')
+
+    example.text = "Foo!"
+
+    expect(page.locator('h1')).to_have_text('Foo!')
+    expect(page.locator('h2')).to_have_text('Foo!')
+
+    example.text = "Baz!"
+
+    expect(page.locator('h1')).to_have_text('Baz!')
+    expect(page.locator('h2')).to_have_text('Foo!')
+
+
+class JSInput(JSComponent):
+
+    text = param.String()
+
+    _esm = """
+    export function render({ model }) {
+      const inp = document.createElement('input')
+      inp.id = 'input'
+      inp.value = model.text
+      inp.addEventListener('change', (event) => {
+        model.text = event.target.value;
+      })
+      return inp
+    }
+    """
+
+
+class ReactInput(ReactComponent):
+
+    text = param.String()
+
+    _esm = """
+    export function render({ model }) {
+      const [text, setText ] = model.useState("text")
+      return (
+        <input
+          id="input"
+          value={text}
+          onChange={e => setText(e.target.value)}
+        />
+      )
+    }
+    """
+
+
+@pytest.mark.parametrize('component', [JSInput, ReactInput])
+def test_gather_input(page, component):
+    example = component(text='Hello World!')
+
+    serve_component(page, example)
+
+    inp = page.locator('#input')
+
+    inp.click()
+    for _ in example.text:
+        inp.press('Backspace')
+    inp.press_sequentially('Foo!')
+    inp.press('Enter')
+
+    wait_until(lambda: example.text == 'Foo!', page)
+
+
+class JSSendEvent(JSComponent):
+
+    clicks = param.Integer(default=0)
+
+    _esm = """
+    export function render({ model }) {
+      const button = document.createElement('button')
+      button.id = 'button'
+      button.onclick = (event) => model.send_event('click', event)
+      return button
+    }
+    """
+
+    def _handle_click(self, event):
+        self.clicks += 1
+
+
+class ReactSendEvent(ReactComponent):
+
+    clicks = param.Integer(default=0)
+
+    _esm = """
+    export function render({ model }) {
+      return <button id="button" onClick={(event) => model.send_event('click', event)}/>
+    }
+    """
+
+    def _handle_click(self, event):
+        self.clicks += 1
+
+
+@pytest.mark.parametrize('component', [JSSendEvent, ReactSendEvent])
+def test_send_event(page, component):
+    button = component()
+
+    serve_component(page, button)
+
+    page.locator('#button').click()
+
+    wait_until(lambda: button.clicks, page)
+
+
+class JSChild(JSComponent):
+
+    child = Child()
+
+    _esm = """
+    export function render({ model }) {
+      const button = document.createElement('button')
+      button.appendChild(model.get_child('child'))
```

Comment:
I hate camel casing :) 

More seriously though, Bokeh internally uses snake case and since we are a Python library and the `model` is a reflection of the Python Parameterized object I do prefer to stick with snake case here.